### PR TITLE
Add apply_filters() before WP_Query calls in Get_Posts class

### PR DESCRIPTION
**Pull request by Koorosh14**
Added 2 new `apply_filters()` before `\WP_Query` calls in `Get_Posts` class so that users can modify filter queries before executing.

### CODE REVIEW

File: includes/core/frontend/Get_Posts.php [hunk 1]
```diff
@@ -395,6 +395,7 @@ public function get_filter_posts() {
 			}
 		}
 
+		$args = apply_filters('ymc_get_filter_posts_query_args_'.$filter_id.'_'.$target_id, $args, $clean_data);
```
Comment (YMC-22) [inline includes/core/frontend/Get_Posts.php:398]: Thanks for the recommendations. But the plugin already has the ability to change custom arguments of the WP_Query object.

File: includes/core/frontend/Get_Posts.php [hunk 2]
```diff
@@ -652,6 +653,7 @@ public function autocomplete_search() {
 			}
 		}
 
+		$args = apply_filters('ymc_autocomplete_search_query_args', $args, $id, $term_ids);
```
Comment (YMC-22) [inline includes/core/frontend/Get_Posts.php:656]: We will consider this in the future as an opportunity to change WP_Query settings.

File: includes/core/frontend/Get_Posts.php (hunk 1 above)
Comment (Koorosh14) [reply includes/core/frontend/Get_Posts.php:398]: Hi there, thank you for this useful plugin. 
I've seen the options to change custom arguments, but as an example there's no way to change the [`operator` parameter](https://developer.wordpress.org/reference/classes/wp_query/#taxonomy-parameters). (Which is exactly the reason that I've added this filter in the first place)

Comment (YMC-22) [reply includes/core/frontend/Get_Posts.php:398]: We will consider your request as soon as possible. Thanks

Comment (YMC-22) [reply includes/core/frontend/Get_Posts.php:398]: You can customize and pass arguments to the WP_query() object. For example, you can explicitly specify the operator inside the function's callbacks (see code below) example:
---
`if ( ! defined( 'YMC_CALLBACK_FUNCTION_WHITELIST' ) ) {

	define( 'YMC_CALLBACK_FUNCTION_WHITELIST', array(
		'my_posts'
	) );
}

function my_posts( $atts ) {

	$term_ids = [];

	if( is_array($atts['term']) && ! empty($atts['term']) ) :

		foreach ( $atts['term'] as $term ) :
			if( 'post_tag' === get_term( $term )->taxonomy) :
				$term_ids[] = (int) $term;
			endif;
		endforeach;

	endif;

	return [
		'relation' => 'AND',
		'tax_query' => [
			[
				'taxonomy' => 'category',
				'field'    => 'id',
				'terms'    => [ 19 ],
				'operator' => 'NOT IN'
			],
			[
				'taxonomy' => 'post_tag',
				'field'    => 'id',
				'terms'    => $term_ids,
				'operator' => 'IN'
			]
		]
	];
}`

there is no particular need to add a filter to override the object arguments.

**Pull request closed without merging**